src/test/unit/algorithms/ConnectionsPerformanceTest.cpp

```diff
@@ -59,52 +58,51 @@ float runTemporalMemoryTest(UInt numColumns, UInt w,   int numSequences,
   // Initialize
 
   TemporalMemory tm;
-  vector<UInt> columnDim;
-  columnDim.push_back(numColumns);
-  tm.initialize(columnDim);
+  tm.initialize( {numColumns} );
 
   cout << (float)timer.getElapsed() << " in " << label << ": initialize"  << endl;
 
-  // Learn
-
-  vector<vector<vector<CellIdx>>> sequences;
-  vector<vector<CellIdx>> sequence;
-  vector<CellIdx> sdr;
-
+  // generate data
+  vector<vector<SDR>> sequences;
   for (int i = 0; i < numSequences; i++) {
+    vector<SDR> sequence;
+    SDR sdr({numColumns});
     for (int j = 0; j < numElements; j++) {
-      sdr = _randomSDR(numColumns, w);
+      const Real sparsity = w / static_cast<Real>(numColumns);
+      sdr.randomize(sparsity, rng);
       sequence.push_back(sdr);
     }
-
     sequences.push_back(sequence);
   }
 
+  // learn
   for (int i = 0; i < 5; i++) {
     for (auto sequence : sequences) {
       for (auto sdr : sequence) {
-        _feedTM(tm, sdr);
-        tm.reset();
+        tm.compute(sdr, true);
+	//TODO get untrained anomaly score here
       }
+      tm.reset();
     }
   }
-
   cout << (float)timer.getElapsed() << " in " << label << ": initialize + learn"  << endl;
 
-  // Test
-
+  // test
   for (auto sequence : sequences) {
     for (auto sdr : sequence) {
-      _feedTM(tm, sdr, false);
-      tm.reset();
+      tm.compute(sdr, false);
+      //TODO get trained (lower) anomaly
```

Comment:
I'd really like to see this check implemented.  I think it would probably catch an errors such as forgeting to call activateDendrites.

Reply:
I will, but preferably not in this PR, maybe after #347 , I'll open an issue to keep track of that.

Reply:
delegated to #354